lib/active_fedora/schema_indexing_strategy.rb

```diff
@@ -0,0 +1,26 @@
+module ActiveFedora
+  ##
+  # An application strategy to also apply solr indexes for each property.
+  # @note If how a field is indexed changes based on property, this would be a
+  #   good place to define that.
+  class SchemaIndexingStrategy
+    # @param [Array<Symbol>] index_types Ways to index all properties.
+    def initialize(index_types=[])
```

Comment:
I'm thinking of the single responsibility principle. To me it seems like this class has two responsibilities: copying properties and supplying indexing hints.  I'm also thinking about the case where I want to index differently based on property name.  What does the API for that look like?